tests/integration/components/select-component-test.js

```diff
@@ -133,3 +151,62 @@ test('It displays the specified component when componentNameForGroupTooltip is p
     'the groupItem argument is passed'
   );
 });
+
+test(
+  'It does not display loading text and does show empty content component when content is empty and loaded',
+  async function(assert) {
+    this.container.register('view:some-view', Ember.View.extend({ templateName: 'some-view' }));
+    this.container.register('template:some-view', hbs`<span data-test-some-component>No results</span>`);
+
+    this.set('content', []);
+
+    await this.render(hbs`
+    {{select-component
+        content=content
+        emptyContentView='some-view'
+    }}`);
+
+    await click(this.helpers.selectChoiceElement);
+
+    assert.ok(!this.helpers.contentLoadingElement, 'Loading ui is not present');
+    assert.ok(!!this.helpers.contentEmptyElement, 'Empty content is present');
+    assert.ok(
+      !!this.helpers.contentEmptyElement.querySelector('[data-test-some-component]'),
+      'Empty content view is present'
+    );
+    assert.ok(!this.helpers.noResultsElement, 'No results message is not present');
+  }
+);
+
+test(
+  'It does not display loading text and does show no results message when content is non-empty and loaded',
+  async function(assert) {
+    this.set('content', ['foo']);
+
+    await this.render(hbs`
+    {{select-component content=content}}`);
+
+    await click(this.helpers.selectChoiceElement);
+    await this.helpers.inputSearchText('no match');
+
+    assert.ok(!this.helpers.contentLoadingElement, 'Loading ui is not present');
+    assert.ok(!this.helpers.contentEmptyElement, 'Empty content is not present');
+    assert.ok(!!this.helpers.noResultsElement, 'No results message is present');
+  }
+);
+
+test(
```

Comment:
@NicholasMohr The below test fails before the change and passes afterwards. The above two tests pass either way, and are really just backfilling tests for existing functionality

Reply:
I would think that the select in this case doesn't have `hasNoResults` and `isLoading` true at the same time, see my overall PR comment

Reply:
I think you're right. In fact, the bug being fixed is actually a case where `isLoading` and `contentIsEmpty` are true at the same time. I'm happy we now have tests for both combinations but the `contentIsEmpty` one is the culprit here.